dashboard/tasks.py

```diff
@@ -0,0 +1,68 @@
+"""
+Periodic task that updates user data.
+"""
+import logging
+from datetime import datetime, timedelta
+from django.conf import settings
+from django.contrib.auth.models import User
+
+from edx_api.client import EdxApi
+from social.apps.django_app.default.models import UserSocialAuth
+
+from backends import utils
+from backends.edxorg import EdxOrgOAuth2
+from dashboard.models import (
+    CachedCertificate,
+    CachedEnrollment,
+)
+from dashboard.api import (
+    get_student_certificates,
+    get_student_enrollments,
+)
+from micromasters.celery import async
+
+
+log = logging.getLogger(__name__)
+
+
+@async.task
+def batch_update_user_data():
+    """
+    Update user data like enrollments, certificates and grades from edX platform.
+    """
+    enrolled_students = CachedEnrollment.objects.filter(
+        last_request__lt=datetime.utcnow() - timedelta(hours=6)
+    ).values_list('user', flat=True)
+
+    certificated_students = CachedCertificate.objects.filter(
+        last_request__lt=datetime.utcnow() - timedelta(hours=6)
+    ).values_list('user', flat=True)
+
+    students = list(set(enrolled_students) | set(certificated_students))
+
+    for user_id in students:
```

Comment:
@noisecapella you are saying, I should add content for loop in ``try`` statement?